### PR TITLE
[Swin] Add Swin SimMIM checkpoints

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR adds 2 checkpoints for Swin Transformer pre-trained using the SimMIM objective (taken from [here](https://github.com/microsoft/Swin-Transformer/blob/main/MODELHUB.md#simmim-pretrained-swin-v1-models)).

They are on the hub: https://huggingface.co/models?other=simmim

It also fixes an important bug in `modeling_swin.py` regarding the window size not being set properly.

### CODE REVIEW

File: src/transformers/models/swinv2/modeling_swinv2.py [hunk 1]
```diff
@@ -1099,8 +1099,8 @@ def forward(
     """,
     SWINV2_START_DOCSTRING,
 )
-# Copied from transformers.models.swin.modeling_swin.SwinForMaskedImageModeling with SWIN->SWINV2,Swin->Swinv2,swin->swinv2,224->256,window7->window8
```
Comment (sgugger) [inline src/transformers/models/swinv2/modeling_swinv2.py:1102]: This means the forward is now decoupled from Swin, is this intended?

**Review by sgugger — APPROVED** [review]
Thanks for adding those!

File: src/transformers/models/swinv2/modeling_swinv2.py (hunk 1 above)
Comment (sgugger) [reply src/transformers/models/swinv2/modeling_swinv2.py:1102]: Can't we just add the checkpoint name in the Copied from statement?

Comment (NielsRogge) [reply src/transformers/models/swinv2/modeling_swinv2.py:1102]: Ok! It doesn't allow for a whole name to be replaced right? So I do it in pieces

Comment (sgugger) [reply src/transformers/models/swinv2/modeling_swinv2.py:1102]: You ca put the whole name, the only thing it doesn't accept are spaces or commas.

Comment (sgugger) [reply src/transformers/models/swinv2/modeling_swinv2.py:1102]: See an example [here](https://github.com/huggingface/transformers/blob/737bff6a36f05761a25444174eab986e028b2b94/src/transformers/models/longformer/tokenization_longformer.py#L115)

**Review by alaradirik — APPROVED** [review]
Thanks for adding the new checkpoints and the fix, everything looks good to me!

**Pull request merged**